### PR TITLE
docs: convert UPGRADING guide to RST to fix table formatting

### LINKED ISSUE #153: Incorrect rendering of tables in the docs
In the [generated docs](https://googleapis.dev/python/datacatalog/latest/UPGRADING.html#supported-python-versions), the tables containing "applicable previous versions" are not rendered correctly, while in the source markdown file [UPGRADING.md](https://github.com/googleapis/python-datacatalog/blob/bb2765dc219580d91ff82611971631b1a62bd41e/UPGRADING.md#supported-python-versions) they are rendered just fine.

### PR DESCRIPTION
**Pull request by tswast**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-datacatalog/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Towards #153🦕

### CODE REVIEW

**Comment by tswast** [comment]
Confirmed with local `nox -s docs` that tables now render correctly:

UPGRADING guide with tables

**Review by meredithslota — APPROVED** [review]
LGTM.

File: docs/index.rst [hunk 1]
```diff
@@ -23,17 +23,6 @@ API Reference
     datacatalog_v1beta1/types
 
 
-Migration Guide
```
Comment (tswast) [inline docs/index.rst:26]: @gcf-owl-bot Nooooooo!

Comment (tswast) [reply docs/index.rst:26]: Mailed https://github.com/googleapis/synthtool/pull/1285 to allow reStructured Text without having to add back an `owlbot.py` file.

**Comment by tswast** [comment]
Thanks @parthea !

**Pull request merged**